File: src/types/equipment-location.ts
export type EquipmentLocation = {
  id: string;
  name: string;
  description: string;
  created_at: string;
  updated_at: string;
};

export type FindEquipmentLocationsDTO = {
  name: string;
  description: string;
};

export type FindOneEquipmentLocationDTO = {
  id: string;
};

export type CreateEquipmentLocationDTO = {
  name: string;
  description: string;
};

export type UpdateEquipmentLocationDTO = {
  id: string;
  name: string;
  description: string;
};

export type DeleteEquipmentLocationDTO = {
  id: string;
};

File: src-tauri/src/dtos/equipment_location.rs
use serde::{Deserialize, Serialize};

#[derive(Deserialize, Serialize, Clone, Debug)]
pub struct NewEquipmentLocationDto {
    pub name: String,
    pub description: String,
}

#[derive(Deserialize, Serialize, Clone, Debug)]
pub struct UpdateEquipmentLocationDto {
    pub id: String,
    pub name: String,
    pub description: String,
}

#[derive(Deserialize, Serialize, Clone, Debug)]
pub struct DeleteEquipmentLocationDto {
    pub id: String,
}

#[derive(Deserialize, Serialize, Clone, Debug)]
pub struct FindEquipmentLocationDto {
    pub name: String,
    pub description: String,
}

#[derive(Deserialize, Serialize, Clone, Debug)]
pub struct FindOneEquipmentLocationDto {
    pub id: String,
}
